src/Marten/Linq/CollectionAnyContainmentWhereFragment.cs

```diff
@@ -19,12 +20,14 @@ public class CollectionAnyContainmentWhereFragment: IWhereFragment
         private readonly MemberInfo[] _members;
         private readonly ISerializer _serializer;
         private readonly SubQueryExpression _expression;
+        private readonly IQueryableDocument _mapping;
 
-        public CollectionAnyContainmentWhereFragment(MemberInfo[] members, ISerializer serializer, SubQueryExpression expression)
+        public CollectionAnyContainmentWhereFragment(MemberInfo[] members, ISerializer serializer, SubQueryExpression expression, IQueryableDocument mapping)
```

Comment:
**Note:** This will be breaking change. I'm not sure if that's fine, as this class should be more internal rather than public. @jeremydmiller @mysticmind @jokokko thoughts?

Reply:
It would be better to ensure the changes are binary compatible.

Reply:
Should I mark this obsolete then?  I'll avoid the breaking change, but IDocumentQuery is needed to build a consistent locator for the query to fix the bug.

Reply:
@barryhagan yes, I think that you can mark it as obsolete with the note that it might be removed in `v4.0`

Reply:
added `[Obsolete()]` to this and other dead constructors